runtime/src/accounts_db.rs

```diff
@@ -1632,27 +1637,77 @@ impl AccountsDB {
         hasher.result()
     }
 
-    fn accumulate_account_hashes(mut hashes: Vec<(Pubkey, Hash)>) -> Hash {
-        let mut sort = Measure::start("sort");
+    fn accumulate_account_hashes(hashes: Vec<(Pubkey, Hash, u64)>) -> Hash {
+        let (hash, ..) = Self::do_accumulate_account_hashes_and_capitalization(hashes, false);
+        hash
+    }
+
+    fn accumulate_account_hashes_and_capitalization(
+        hashes: Vec<(Pubkey, Hash, u64)>,
+    ) -> (Hash, u64) {
+        let (hash, cap) = Self::do_accumulate_account_hashes_and_capitalization(hashes, true);
+        (hash, cap.unwrap())
+    }
+
+    fn do_accumulate_account_hashes_and_capitalization(
+        mut hashes: Vec<(Pubkey, Hash, u64)>,
+        calculate_cap: bool,
+    ) -> (Hash, Option<u64>) {
+        let mut sort_time = Measure::start("sort");
         hashes.par_sort_by(|a, b| a.0.cmp(&b.0));
-        sort.stop();
-        let mut hash_time = Measure::start("hash");
+        sort_time.stop();
 
-        let fanout = 16;
+        let mut sum_time = Measure::start("cap");
+        let cap = if calculate_cap {
```

Comment:
any idea how much this adds to the time?

Reply:
Ok, I see the other benchmarks have it at 700us for 80k signatures and seems to add about 5%. If so, that's not to bad.